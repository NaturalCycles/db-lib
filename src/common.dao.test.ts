import { mockTime } from '@naturalcycles/test-lib'
import { CommonDao } from './common.dao'
import { InMemoryDB } from './inMemory.db'
import { mockTestData, TEST_KIND, TestKindBM, testKindBMSchema } from './test/testData.mock'

const ID = 'randomDatastoreService1'

const db = new InMemoryDB()

const createDao = () => {
  return new CommonDao<TestKindBM>({
    table: TEST_KIND,
    db,
    dbmSchema: testKindBMSchema,
    bmSchema: testKindBMSchema,
  })
}

beforeEach(() => {
  jest.restoreAllMocks()
  mockTime()
  jest.spyOn(require('@naturalcycles/nodejs-lib'), 'stringId').mockImplementation(() => ID)
  db.reset()
})

// todo: mock data set, test all methods one by one

test('full test', async () => {
  const dao = createDao()

  expect(await dao.getById()).toBeUndefined()
  expect(await dao.getById('id1')).toBeUndefined()
  expect(await dao.getByIds(['id1', 'id2', 'idX'])).toEqual([])
  expect(await dao.getBy('c', true)).toEqual([])

  await dao.saveBatch(mockTestData())

  expect(await dao.getById()).toBeUndefined()
  expect(await dao.getById('id1')).toMatchSnapshot()
  expect(await dao.getByIds(['id1', 'id2', 'idX'])).toMatchSnapshot()
  expect(await dao.getBy('c', true)).toMatchSnapshot()
})
