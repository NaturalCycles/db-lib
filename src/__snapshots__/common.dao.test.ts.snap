// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full test 1`] = `
Object {
  "a": "a1",
  "b": "b1",
  "c": false,
  "created": 1529539200,
  "id": "id1",
  "updated": 1529539200,
}
`;

exports[`full test 2`] = `
Array [
  Object {
    "a": "a1",
    "b": "b1",
    "c": false,
    "created": 1529539200,
    "id": "id1",
    "updated": 1529539200,
  },
  Object {
    "a": "a2",
    "b": "b2",
    "c": true,
    "created": 1529539200,
    "id": "id2",
    "updated": 1529539200,
  },
]
`;

exports[`full test 3`] = `
Array [
  Object {
    "a": "a2",
    "b": "b2",
    "c": true,
    "created": 1529539200,
    "id": "id2",
    "updated": 1529539200,
  },
  Object {
    "a": "a4",
    "b": "b4",
    "c": true,
    "created": 1529539200,
    "id": "id4",
    "updated": 1529539200,
  },
]
`;
